### DISCUSSION TITLE
How to add custom chain from rainbowkit chain dropdown to metamask (in case it doesn't have it yet)?

### DISCUSSION THREAD

**Discussion by calebwick** [rainbow-me/rainbowkit#1537] [Q&A] [answered]
`<RainbowKitProvider modalSize="compact" chains={chains} theme={myTheme}>
`
For example, when I put a custom chain in the chains array, when I click the dropdown and metamask doesn't have it, it will throw console error: 
`MetaMask - RPC Error: Unrecognized chain ID "0xa"`

Is this expected?

**Comment by DanielSinclair** [comment, marked as answer]
@calebwick Do you mean that you selected the unsupported network from the dropdown? If so, yes, that is currently expected. This error is returned directly from MetaMask's RPC provider. Other wallets will show a nicer "Unsupported Network" prompt and handle this better. We could improve the UX for RainbowKit as well and show an error. Do you have any ideas?

**Reply by calebwick** [reply]
Is it possible to have like a function that we can pass in for when the error is caught? So we can add the prompt/toast to the user.
Or perhaps call the 'wallet_addEthereumChain' directly?
